### ISSUE TITLE
NextJS 15 with chakra UI 3 (and 2 also)

### ISSUE THREAD

**Issue by ivan-dalmet** [chakra-ui/chakra-ui#8970]
### Description

Getting hydratation error even with `suppressHydratationWarning` in NextJS 15.
![image](https://github.com/user-attachments/assets/63f66869-5bad-47c3-bf4c-b28c5a03390f)

### Link to Reproduction

https://codesandbox.io/p/devbox/77hrfw

### Steps to reproduce

1. Open the preview
2. You will get an hydratation error

### Chakra UI Version

3.0.0-next.32 (but same error with 2.x)

### Browser

Ark

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

Maybe the `suppressHydratationWarning` does not work anymore in NextJS 15? or React 19?

**Comment by lcswillems** [comment]
@segunadebayo In the doc, it says to add `suppressHydratationWarning` to hide the warning, but we should not hide the warning. If the warning is displayed, it is because Chakra 3 doesn't something incorrect, so instead Chakra 3 should be fixed?

**Comment by ivan-dalmet** [comment]
@lcswillems yes but also the issue here is that suppressHydratationWarning doesn't work 😟
So following the docs does not work

**Comment by ivan-dalmet** [comment]
By the way, the error doesn't seem to show at each reload 😥

**Comment by segunadebayo** [comment]
This is probably related to https://github.com/pacocoursey/next-themes or React 19

@lcswillems here's what next-themes recommends. It's an intentional decision; the prop exists in react for edge cases like this.
![CleanShot 2024-10-25 at 17 32 06@2x](https://github.com/user-attachments/assets/aab5c94e-ac57-43b0-b0d6-78d47751273e)

**Comment by baptisteArno** [comment]
Exact same issue here even with the `suppressHydrationWarning` workaround (React 18). Reverting to Next v14 fixes it

**Comment by unishigure** [comment]
Hi. I have the same problem in Next.js v15.0.1.
This problem only occurs when `--turbopack` option in my environment.

**Comment by PatrickRogg** [comment]
Adding an emotion cache solved it for me. From here: https://github.com/emotion-js/emotion/issues/2928#issuecomment-1293012737

```
'use client';

import { ChakraProvider } from '@chakra-ui/react';
import { ColorModeProvider } from '../components/ui/color-mode';
import { system } from '../style-system';
import { EmotionCacheProvider } from './emotion-cache';

export function DesignThemeProvider(props: { children: React.ReactNode }) {
  return (
    <EmotionCacheProvider>
      <ChakraProvider value={system}>
        <ColorModeProvider defaultTheme="light">{props.children}</ColorModeProvider>
      </ChakraProvider>
    </EmotionCacheProvider>
  );
}
```

```
'use client';

import React, { useState } from 'react';
import { useServerInsertedHTML } from 'next/navigation';
import { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export function EmotionCacheProvider({ children }: { children: React.ReactNode }) {
  const [cache] = useState(() => {
    const cache = createCache({ key: 'chakra', prepend: true });
    // @ts-ignore
    cache.compat = true; // Ensure the cache is compatible with Emotion 10
    return cache;
  });

  useServerInsertedHTML(() => (
    <style
      data-emotion={`${cache.key} ${Object.keys(cache.inserted).join(' ')}`}
      dangerouslySetInnerHTML={{
        __html: Object.values(cache.inserted).join(' ')
      }}
    />
  ));

  return <CacheProvider value={cache}>{children}</CacheProvider>;
}
```

**Comment by sayshark75** [comment]
Hey @segunadebayo , is this issue resolved? i didn't face any such issue of hydration, by following the app router instruction in docs.
also no hydration issue in the codesandbox link. Thanks.

**Comment by skylarmb** [comment]
I can confirm this only happens with `--turbopack`

**Comment by segunadebayo** [comment]
Given this issue only happens in `--turbopack`, I'm not sure what we can do about it.

Moving it to a discussion until we have a definite plan of action.
